### DISCUSSION TITLE
Possible to import video (MP4 etc.) with green background and chroma-key the green as "transparent"

### DISCUSSION THREAD

**Discussion by webhype** [keiwando/toonsquid#390] [General]
I am trying to get an animated puppet from Adobe Express (or Character Animator) into ToonSquid. What's the best practice to remove the background on imported assets? Ideal would be a chroma key feature that declares everything that's green (or a choosable color), to be transparent instead. CapCut style. If this feature already exists, I didn't find it?! Thanks!

**Comment by Cat-Butler** [comment]
That's the sort of thing you're more likely to be able to do in a video editing app (like CapCut), rather than an animation app like ToonSquid (just as you haven't mentioned compositing ToonSquid green screen output in Character Animator). ToonSquid is another choice for creating that animated puppet part of your work.

You can take then your animated puppet video from Character Animator, your animation video output from ToonSquid, and composite them (and add sound, effects, titling and credits, etc) in the video editing app, whether in a desktop product, Adobe or otherwise, or in an iPad video editor.

There are some decent video editing apps available on the iPad. There is a tendency for them to be subscription-based, although several have (somewhat hefty) lifetime licence options. 
LumaFusion is fairly well known and has a once-off reasonable price with a subscription for added content, Davince Resolve recently became available for M1 or better iPads with a free base version and a paid Studio upgrade, InShot seeems reasonable, NodeVideo is currently poorly documented but has some strong effects options.

**Reply by webhype** [reply]
Thanks for your reply; you're not wrong, and you're kind of making my point... to me, Adobe Ch and ToonSquid are not to be used "one after the other" in a sequential workflow, like ToonSquid -> AfterEffects -> Premiere Pro.

Instead, they are strong "peers" with different strengths, complementing each other. Specifically, I want to use Adobe Ch to save myself time animating the face of my character, but want to avoid the awkward "marionette-like" puppet body, by drawing my own body to escape the limitations of Adobe Ch.

Exactly for that reason, I would appreciate great interoperability and would like to import my puppet's animated face to plant it on a body that I animate in ToonSquid. It is exactly because they are both made to "create smaller parts that flow into a greater whole", that that there should be a free back-and-forth between the two programs.

ToonSquid goes already out of its way to have tons of Photoshop-inspired blend modes; if there were just one more called "chroma key masked" or something like that, it would allow almost unlimited interoperability with Adobe Ch and other "peer" programs that you still need here and there, to get the result that you want.

**Reply by Cat-Butler** [reply]
Does Character Animator support outputting a PNG image sequence with transparency?
That's often a better interoperability format for moving between apps if your workflow supports it.

ToonSquid should import an image sequence into the main timeline without problems.
https://toonsquid.com/handbook/layers/video/

You could possibly then convert that layer to a symbol if it's more convenient.

**Reply by Cat-Butler** [reply]
The other possibility is to animate the body layer over the head layer in ToonSquid (no green screen), export the body layer separately, then composite the head and the body animations in a video editor as previously mentioned, with the head layer on top.

**Comment by webhype** [comment]
Excellent yes, it seems like Adobe Character Animator can export PNG sequences... https://helpx.adobe.com/adobe-character-animator/using/export.html ... I will try that out, thank you! ❤️

**Reply by webhype** [reply]
Having said all that, I still think a chroma key (green screen) blend mode would be mega cool and again, save a lot of time!
